src/SprykerSdk/Zed/AopSdk/Business/OpenApi/Builder/OpenApiCodeBuilder.php

```diff
@@ -0,0 +1,732 @@
+        }
+
+        return $this->formatResponseProperties($this->getPropertiesFromReference($schema));
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return array
+     */
+    protected function formatPropertiesToArrayOfClassInstance(string $className): array
+    {
+        $refClass = str_replace('Attributes', '', $className);
+
+        return [$this->inflector->pluralize($refClass) => $refClass . '[]:' . $this->inflector->camelize($refClass)];
+    }
+
+    /**
+     * @param iterable $properties
+     *
+     * @return array
+     */
+    protected function formatResponseProperties(iterable $properties): array
+    {
+        $response = [];
+        foreach ($properties as $key => $schemaObject) {
+            if (isset($schemaObject->type) && $schemaObject->type !== 'array') {
+                $response[$key] = $schemaObject->type;
+
+                continue;
+            }
+
+            if (!isset($schemaObject->items)) {
+                continue;
+            }
+
+            if (isset($schemaObject->items->type)) {
+                $response[$key] = 'array[]:' . $schemaObject->items->type;
+
+                continue;
+            }
+
+            if (isset($schemaObject->items->properties) && isset($schemaObject->items->properties['type'])) {
+                $response[$key] = 'array[]:' . $schemaObject->items->properties['type']->type;
+
+                continue;
+            }
+        }
+
+        return $response;
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Schema $schema
+     *
+     * @return string
+     */
+    protected function getClassNameFromSchema(Schema $schema): string
+    {
+        if ($schema->getDocumentPosition()) {
+            $referencePath = $schema->getDocumentPosition()->getPath();
+
+            return end($referencePath);
+        }
+
+        return '';
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Reference $reference
+     *
+     * @return string
+     */
+    protected function getClassNameFromReference(Reference $reference): string
+    {
+        if ($reference->getDocumentPosition()) {
+            $referencePath = $reference->getDocumentPosition()->getPath();
+
+            return end($referencePath);
+        }
+
+        return '';
+    }
+
+    /**
+     * @param array $parameters
+     *
+     * @return array
+     */
+    protected function preparePropertyNameForCommand(array $parameters): array
+    {
+        $parsedProperties = [];
+        foreach ($parameters as $key => $value) {
+            $parsedProperties[] = "{$key}:{$value}";
+        }
+
+        return $parsedProperties;
+    }
+
+    /**
+     * @param string $organization
+     * @param array $parseProperties
+     *
+     * @return array
+     */
+    protected function getTransferDefinitionSprykCommands(string $organization, array $parseProperties): array
+    {
+        $commandLines = [];
+
+        foreach ($parseProperties as $operations) {
```

Comment:
This nesting would not be needed when `$transferDefinitions` is used properly.

```
[
    'transferNameA' => [ transfer definition a ],
    'transferNameB' => [ transfer definition b ],
]
```

We only need operations and alike with the next step of code generators.

Reply:
FYI: The next part we want to use this builder for will be for generating the Glue code we need for a resource. There we will need more information than we need for the transfer objects.

We will need something like this next:

```
$resourceMethodResponses = [
    '/foo-bars' => [
        'moduleName' => 'FooBars',
        'controllerName' => 'FooBarsResourceController',
        'resourceDataObject' => 'FooBar',
        'isBulk' => false,
        'httpMethod' => 'get',
        'httpResponseCodes' => [
            200,
        ],
        ....
    ]
];
```
Just keep that in mind. This is not final.